x-pack/plugins/ml/server/models/results_service/results_service.js

```diff
@@ -0,0 +1,255 @@
+        }
+      });
+    }
+
+    const resp = await callWithRequest('search', {
+      index: ML_RESULTS_INDEX_PATTERN,
+      size: maxRecords !== undefined ? maxRecords : 500,
+      body: {
+        query: {
+          bool: {
+            filter: [
+              {
+                query_string: {
+                  query: 'result_type:record',
+                  analyze_wildcard: false
+                }
+              },
+              {
+                bool: {
+                  must: boolCriteria
+                }
+              }
+            ]
+          }
+        },
+        sort: [
+          { record_score: { order: 'desc' } }
+        ]
+      }
+    });
+
+    const tableData = { anomalies: [], interval: 'second' };
+    if (resp.hits.total !== 0) {
+      let records = [];
+      resp.hits.hits.forEach((hit) => {
+        records.push(hit._source);
+      });
+
+      // Sort anomalies in ascending time order.
+      records = _.sortBy(records, 'timestamp');
+      tableData.interval = aggregationInterval;
+      if (aggregationInterval === 'auto') {
+        // Determine the actual interval to use if aggregating.
+        const earliest = moment(records[0].timestamp);
+        const latest = moment(records[records.length - 1].timestamp);
+
+        const daysDiff = latest.diff(earliest, 'days');
+        tableData.interval = (daysDiff < 2 ? 'hour' : 'day');
+      }
+
+      tableData.anomalies = buildAnomalyTableItems(records, tableData.interval);
+
+      // Load examples for any categorization anomalies.
+      const categoryAnomalies = tableData.anomalies.filter(item => item.entityName === 'mlcategory');
+      if (categoryAnomalies.length > 0) {
+        tableData.examplesByJobId = {};
+
+        const categoryIdsByJobId = {};
+        categoryAnomalies.forEach((anomaly) => {
+          if (!_.has(categoryIdsByJobId, anomaly.jobId)) {
+            categoryIdsByJobId[anomaly.jobId] = [];
+          }
+          if (categoryIdsByJobId[anomaly.jobId].indexOf(anomaly.entityValue) === -1) {
+            categoryIdsByJobId[anomaly.jobId].push(anomaly.entityValue);
+          }
+        });
+
+        const categoryJobIds = Object.keys(categoryIdsByJobId);
+        await Promise.all(categoryJobIds.map(async (jobId) => {
+          const examplesByCategoryId = await getCategoryExamples(jobId, categoryIdsByJobId[jobId], maxExamples);
+          tableData.examplesByJobId[jobId] = examplesByCategoryId;
+        }));
+      }
+
+    }
+
+    return tableData;
+
+  }
+
+
+  // Obtains the categorization examples for the categories with the specified IDs
+  // from the given index and job ID.
+  // Returned response consists of a list of examples against category ID.
+  async function getCategoryExamples(jobId, categoryIds, maxExamples) {
+    const resp = await callWithRequest('search', {
+      index: ML_RESULTS_INDEX_PATTERN,
+      size: 500,    // Matches size of records in anomaly summary table.
```

Comment:
nit: could be a constant in the beginning of the file and also replace the other occurence.